src/transforms/filter.js

```diff
@@ -101,7 +102,17 @@ exports.attributes = {
             '*value* is expected to be an array with as many items as',
             'the desired set elements.'
         ].join(' ')
-    }
+    },
+    preservegaps: {
+        valType: 'boolean',
+        dflt: false,
+        description: [
+            'Determines whether or not gaps in data arrays produced by the filter operation',
+            'are preserved or not.',
+            'Setting this to *true* might be useful when plotting a line chart',
+            'with `connectgaps` set to *true*.'
```

Comment:
wait, do you mean with `connectgaps` set to `false`? `connectgaps: true` would run connect right over the gaps you just preserved...